lib/bitstyles_phoenix/bitstyles.ex

```diff
@@ -101,52 +118,83 @@ defmodule BitstylesPhoenix.Bitstyles do
         "u-overflow-x-auto" -> "u-overflow--x"
         "u-overflow-y-auto" -> "u-overflow--y"
+        "u-version-4" -> "u-version-2"
         _ -> class
       end
 
-    classname(mapping, "4.0.0")
+    downgrade_classname(mapping, target_version, {2, 0, 0})
   end
 
-  def classname(class, version) when version >= "1.5.0" do
+  defp downgrade_classname(class, target_version, current_version)
+       when should_downgrade_from({2, 0, 0}, target_version, current_version) do
     mapping =
       case class do
         "u-flex-shrink-" <> number -> "u-flex__shrink-#{number}"
         "u-flex-grow-" <> number -> "u-flex__grow-#{number}"
         "u-flex-wrap" -> "u-flex--wrap"
         "u-flex-col" -> "u-flex--col"
+        "u-version-2" -> "u-version-1-5"
         _ -> class
       end
 
-    classname(mapping, "2.0.0")
+    downgrade_classname(mapping, target_version, {1, 5, 0})
   end
 
-  def classname(class, version) when version >= "1.3.0" do
+  defp downgrade_classname(class, target_version, current_version)
+       when should_downgrade_from({1, 5, 0}, target_version, current_version) do
     mapping =
       case class do
         "u-grid-cols-" <> number -> "u-grid--#{number}-col"
         "u-col-span-" <> number -> "u-grid__col-span-#{number}"
         "u-col-start-" <> number -> "u-grid__col-#{number}"
+        "u-version-1-5" -> "u-version-1-3"
         _ -> class
       end
 
-    classname(mapping, "1.5.0")
+    downgrade_classname(mapping, target_version, {1, 3, 0})
   end
 
-  def classname(_class, version) do
+  defp downgrade_classname(_class, target_version, current_version)
+       when should_downgrade_from({1, 3, 0}, target_version, current_version) do
     raise("""
-    The version #{version} of bitstyles is not supported. The helpers will produce incorrect classes.
+    The version #{version_to_string(target_version)} of bitstyles is not supported. The helpers will produce incorrect classes.
     Please upgrade bitsyles and set the `bitsyles_version` to the updated version.
     """)
   end
 
   def version do
+    version_to_tuple(version_string())
+  end
+
+  def version_string do
     bitstyles_version_override = Process.get(:bitstyles_phoenix_bistyles_version)
 
     bitstyles_version_override ||
       Application.get_env(:bitstyles_phoenix, :bitstyles_version, @default_version)
   end
 
   def default_version do
+    version_to_tuple(@default_version)
+  end
+
+  def default_version_string do
     @default_version
   end
+
+  defp version_to_tuple(version) when is_tuple(version), do: version
+
+  defp version_to_tuple(version) when is_binary(version) do
+    version
+    |> String.split(".")
+    |> Enum.map(&String.to_integer/1)
+    |> List.to_tuple()
+  end
+
+  defp version_to_string(version) when is_binary(version), do: version
+
+  defp version_to_string(version) when is_tuple(version) do
+    version
+    |> Tuple.to_list()
+    |> Enum.map_join(".", &to_string/1)
+  end
```

Comment:
I think with the increased complexity in those helpers, should we extract this into 
```
BitstylesPhoenix.Bitstyles.Version
```
Then you can just implement `to_string` and `to_tuple` :)

Reply:
Is this was you had in mind [`6c35953` (#134)](https://github.com/bitcrowd/bitstyles_phoenix/pull/134/commits/6c35953e10265d72a3b46bdb551fb51b3a1bcbbc) ? Somehow `Bitstyles.Version.version()` and `Bitstyles.Version.default_version()` reads weird, but I wasn't sure how to name those functions